packages/next-international/src/common/create-t.ts

```diff
@@ -15,9 +15,17 @@ export function createT<Locale extends BaseLocale, Scope extends Scopes<Locale>
   scope: Scope | undefined,
 ) {
   const { localeContent, fallbackLocale } = context;
-
+  // If there is no localeContent (e.g. on initial render on the client-side), we use the fallback locale
+  // otherwise, we use the fallback locale as a fallback for missing keys in the current locale
+  const content =
+    fallbackLocale && typeof localeContent === 'string'
+      ? fallbackLocale
+      : {
+          ...(fallbackLocale ?? {}),
```

Comment:
on a second thought, this is not a real merge and should probably be a deep nested merge

Reply:
There is not need to deep merge, because both `fallbackLocale` and `localeContent` are already flattened and are a `Record<string, string>` at this point.

Instead of a spread, you should use `Object.assign` which will be a lot faster when having lots of keys:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals